docs/adr/0003-delivery-channels-templating.md

```diff
@@ -0,0 +1,92 @@
+# Templating engine for different notification types
+
+## Status
+
+Proposed
+
+## Context
+
+This ADR proposes a method on how to implement custom templating when a certain notification is sent through a communication channels (SMS, email, web push notifications etc).
+
+In a typical notification system, an object will be applied to a basic template to provide an end user with information targeting them. A lot of the basic design/layout of the template will be quite standard, but there will be places for values that will need to be inserted to customise it for the said end user. An example of this would be an birthday email notification, where a lot of the design and formatting of the email/subject will already be very well defined, and then insertion of custom data (eg. email id, name, etc ) is what will be needed by a templater.
+
+The content of this ADR will work off themes from the previous ADR (0002-delivery-channel-creation-and-integration.md).
+
+## Decision
+
+A notification having necessary information needs to be translated into a certain communication object (SMS, Email, etc) before it can be sent via the desired delivery channel (SendGrid, Twillio, Airship).
+
+An example of this would be a SMS, which needs the essential information to be available for it to be converted into an SMS object before being sent to the end user.
+
+```json
```

Comment:
No it isn't. It looks more like C#. Might be better if it were JSON though.

Reply:
Yes. Changed to C# as these objects would be passed into different delivery channel functions.